### PR TITLE
Fix/#1185 corrected type hinting ExponentialSmoothing model

### PR DESCRIPTION
**Pull request by rijkvandermeulen**
Fixes mistake in type hinting ExponentialSmoothing model

### Summary

`seasonal: Optional[SeasonalityMode]` iso `seasonal: Optional[ModelMode]`

### Other Information

Potentially useful to add mypy to pipeline and pre-commit hooks?

### CODE REVIEW

**Comment by rijkvandermeulen** [comment]
Pipeline seems to break when executing CodeCov on macos (but not on ubuntu). Not sure exactly what's going on. Have you seen this before?

File: darts/models/forecasting/exponential_smoothing.py [hunk 1]
```diff
@@ -34,7 +34,10 @@ def __init__(
         <https://www.statsmodels.org/stable/generated/statsmodels.tsa.holtwinters.ExponentialSmoothing.html>`_;
         we refer to this link for the original and more complete documentation of the parameters.
 
-        `model_mode` must be a ``ModelMode`` Enum member. You can access the Enum with ``from darts import ModelMode``.
+        `model_mode` must be a ``ModelMode`` Enum member. You can access the Enum with
```
Comment (hrzn) [inline darts/models/forecasting/exponential_smoothing.py:37]: Do we still need this line?

File: darts/models/forecasting/exponential_smoothing.py [hunk 2]
```diff
@@ -34,7 +34,10 @@ def __init__(
...
+         ``from darts.utils.utils import ModelMode``.
+        `season_mode` must be a ``SeasonalityMode`` Enum member. You can access the Enum with
```
Comment (hrzn) [inline darts/models/forecasting/exponential_smoothing.py:39]: Shouldn't it be
```suggestion
        `seasonal` must be a ``SeasonalityMode`` Enum member. You can access the Enum with
```

**Comment by hrzn** [comment]
> Pipeline seems to break when executing CodeCov on macos (but not on ubuntu). Not sure exactly what's going on. Have you seen this before?

Sometimes the codecov upload fails, I've just relaunched it.

File: darts/models/forecasting/exponential_smoothing.py (hunk 2 above)
Comment (rijkvandermeulen) [reply darts/models/forecasting/exponential_smoothing.py]: Indeed! Changed it.

File: darts/models/forecasting/exponential_smoothing.py (hunk 1 above)
Comment (rijkvandermeulen) [reply darts/models/forecasting/exponential_smoothing.py:37]: Yes would say so as the ModelMode enum is still used for the trend argument in the ES model

Comment (hrzn) [reply darts/models/forecasting/exponential_smoothing.py:37]: Then I think we should write `trend` must be a ``ModelMode`` ... no?

Comment (rijkvandermeulen) [reply darts/models/forecasting/exponential_smoothing.py]: Jup of course - apologies for being sloppy here. Fixed it now.

**Review by hrzn — APPROVED** [review]
thanks!

**Pull request merged**